src/main/java/de/wejago/hichi2influx/service/JsonSubscriber.java

```diff
@@ -0,0 +1,72 @@
+package de.wejago.hichi2influx.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.influxdb.client.domain.WritePrecision;
+import com.influxdb.client.write.Point;
+import de.wejago.hichi2influx.config.Device;
+import de.wejago.hichi2influx.repository.InfluxDbRepository;
+import java.time.Instant;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.map.HashedMap;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class JsonSubscriber extends Subscriber{
+    private final ObjectMapper objectMapper;
+    private final InfluxDbRepository influxDbRepository;
+    private final Device device;
+
+    @Override public void messageArrived(String s, MqttMessage mqttMessage) throws Exception {
+        try {
+            String receivedMessage = new String(mqttMessage.getPayload());
+
+            if(device != null && device.getOnly_match() != null && receivedMessage.contains(device.getOnly_match())) {
+                Map<String, Object> deviceToPointProperties = readDataFromMqttMessage(receivedMessage);
```

Comment:
Better read the data to a JsonTree, then you can walk it with the mapping.
Have a look here: https://attacomsian.com/blog/processing-json-spring-boot#json-file-into-a-tree-model